app-office/drawio-desktop/drawio-desktop-22.1.2.ebuild

```diff
@@ -0,0 +1,115 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+MY_PN="draw.io"
+MY_NODE_N="node-modules"
+MY_NODE_D="node_modules"
+MY_NODE_PV="${PV}"
+DRAWIO_V="22.1.3"
+
+inherit desktop xdg
+
+DESCRIPTION="draw.io diagramming and whiteboarding desktop app"
+HOMEPAGE="https://www.drawio.com/"
+
+SRC_URI="
+	https://github.com/jgraph/drawio-desktop/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz
+	https://github.com/jgraph/drawio/archive/refs/tags/v${DRAWIO_V}.tar.gz
+	https://github.com/MocioF/distfiles/raw/main/${PN}-${MY_NODE_N}-${PV}-${DRAWIO_V}.tar.xz
+"
+
+LICENSE="Apache-2.0"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+RDEPEND="
+	>=net-libs/nodejs-16[ssl]
+	>=dev-libs/nss-3
+	app-accessibility/at-spi2-core
+	net-print/cups
+	x11-libs/libdrm
+	x11-libs/gtk+:3
+	x11-libs/libxkbcommon
+	media-libs/alsa-lib
+"
+
+MY_NODE_DIR="${S}/${MY_NODE_D}/"
+
+src_prepare() {
+	default
+	# We will use pre-generated npm stuff.
+	mv "${WORKDIR}/${MY_NODE_D}" "${MY_NODE_DIR}" || die "couldn't move node_modules"
+
+	# Remove unneeded modules with Pre-stripped files
+	rm -r "${MY_NODE_DIR}/7zip-bin" || die "couldn't remove 7zip-bin module"
+	rm -r "${MY_NODE_DIR}/app-builder-bin" || die "couldn't remove app-builder-bin module"
+
+	chmod 755 -R "${MY_NODE_DIR}"
+
+	# Move drawio sources.
+	rm -r "${WORKDIR}/${P}/drawio" || die "couldn't remove old drawio dir"
+	mv "${WORKDIR}/drawio-${DRAWIO_V}" "${WORKDIR}/${P}/drawio" || die "couldn't move drawio sources"
+
+	cp "${FILESDIR}/drawio-desktop-edit.sh" "${WORKDIR}/${P}/drawio-desktop-edit.sh" || die "couldn't copy sh file"
```

Comment:
The `drawio-desktop-edit.sh` script is small enough that it can be created by a bash here doc in `src_install()`. This avoids having to copy'ing it over and sed'ing it.